### ISSUE TITLE
CS2: How to get crc-map code?

### ISSUE THREAD

**Issue by NintyS** [markus-wa/demoinfocs-golang#483]
How to get CRC code of map in CS2 version of the library?
The msgs2.CSVCMsg_ServerInfo don't have a GetMapCrc() function as I see

**Comment by dankotov** [comment]
@NintyS were you ever able to figure out?

**Comment by NintyS** [comment]
Nah, probably you can't. But I get from CS:GO the CRC codes and they works so for now I'm good.
```go
var (
	Mirage   uint32 = 1936772555
	Anubis   uint32 = 3934213780
	Nuke     uint32 = 4081488007
	Inferno  uint32 = 3201302029
	Ancient  uint32 = 4262714479
	Overpass uint32 = 2863184063
	Vertigo  uint32 = 970160341
)
```

**Comment by dankotov** [comment]
Understood! Thanks for the list. However, how do you get the map name from the demo to get the corresponding crc? `header.MapName` returning `nil` for me.

**Comment by NintyS** [comment]
Sorry for delay, I didn't get notification until I open mail. AFAIK header is deprecated / returns nothing because new header is different than old one.

This is my code:
`
var mapName string
	var mapMetaData ex.Map

	p.RegisterNetMessageHandler(func(msg *msgs2.CSVCMsg_ServerInfo) {

		var mapCode uint32 = 0

		if strings.Contains(msg.GetMapName(), "mirage") {
			mapCode = Mirage
		}

		if strings.Contains(msg.GetMapName(), "overpass") {
			mapCode = Overpass
		}

		if strings.Contains(msg.GetMapName(), "inferno") {
			mapCode = Inferno
		}

		if strings.Contains(msg.GetMapName(), "vertigo") {
			mapCode = Vertigo
		}

		if strings.Contains(msg.GetMapName(), "anubis") {
			mapCode = Anubis
		}

		if strings.Contains(msg.GetMapName(), "nuke") {
			mapCode = Nuke
		}

		if strings.Contains(msg.GetMapName(), "ancient") {
			mapCode = Ancient
		}

		mapMetaData = ex.GetMapMetadata(msg.GetMapName(), mapCode)

		fmt.Println(mapMetaData)

		mapName = msg.GetMapName()
	})`

**Comment by NintyS** [comment]
It's shit but for now this is only thing I can do I guess. Maybe I will post my repo with code as a example for people.

**Comment by dankotov** [comment]
Understood! Thanks for sharing

**Comment by dankotov** [comment]
Out of curiosity, is there any reason you dont do it like this?

`
var mapCodes = map[string]uint32{
	"de_mirage":   1936772555,
	"de_anubis":   3934213780,
	"de_nuke":     4081488007,
	"de_inferno":  3201302029,
	"de_ancient":  4262714479,
	"de_overpass": 2863184063,
	"de_vertigo":  970160341,
}

var mapMetadata ex.Map

p.RegisterNetMessageHandler(func(msg *msgs2.CSVCMsg_ServerInfo) {
	mapName := msg.GetMapName() // change if you want mapName to be available somewhere else as well
	mapMetadata = ex.GetMapMetadata(mapName, mapCodes[mapName])
})
`

**Comment by NintyS** [comment]
Yes, because I forgot that Key-Value dictionaries exists XD

**Comment by markus-wa** [comment]
Sorry, closed this prematurely - still no idea how to get this or if it's even possible for cs2

**Comment by markus-wa** [comment]
Cs2 itself does not support playing back demos that are incompatible with the current map version - so chances the crc to download the correct one isn't there

**Comment by markus-wa** [comment]
The question remaining is how does the game determine that a map is incompatible to show this error

**Comment by markus-wa** [comment]
closing as I don't think we will find an answer to this
